### PR TITLE
Format code with black

### PR DESCRIPTION
**Pull request by virtuald**
The time has come.

### CODE REVIEW

File: docs/gensidebar.py [hunk 1]
```diff
@@ -5,86 +5,91 @@
...
+    lines = ["", ".. DO NOT MODIFY! THIS PAGE IS AUTOGENERATED!", ""]
+
     def toctree(name):
-        lines.extend(['.. toctree::',
-                      '    :caption: %s' % name,
-                      '    :maxdepth: 2',
-                      ''])
-    
+        lines.extend(
+            [".. toctree::", "    :caption: %s" % name, "    :maxdepth: 2", ""]
```
Comment (ariovistus) [inline docs/gensidebar.py]: does black require these to be on one line? I think it looks better when each item is on its own line

Comment (virtuald) [reply docs/gensidebar.py]: Black isn't really configurable, except by turning formatting off. I agree this part is ugly, but it's not really user-facing so I'm fine with that.

File: wpilib/tests/test_drive.py [hunk 2]
```diff
@@ -248,16 +259,21 @@ def check_curvature(wpimock, drive_diff, y, rotation, isQuickTurn):
...
-    drive_diff.rightMotor.set.assert_called_once_with(-rightMotorSpeed * drive_diff.maxOutput)
+    drive_diff.leftMotor.set.assert_called_once_with(
+        leftMotorSpeed * drive_diff.maxOutput
+    )
+    drive_diff.rightMotor.set.assert_called_once_with(
+        -rightMotorSpeed * drive_diff.maxOutput
+    )
 
 
+# fmt: off
```
Comment (ariovistus) [inline wpilib/tests/test_drive.py]: why is this one needed?

File: docs/gensidebar.py (hunk 1 above)
Comment (auscompgeek) [reply docs/gensidebar.py]: The alternative is to call `lines.append` a bunch of times, I guess. Follow your heart, I suppose.

File: wpilib/tests/test_drive.py (hunk 2 above)
Comment (ariovistus) [reply wpilib/tests/test_drive.py]: I know. It looks like the code you're ignoring is the test cases on 280 - 289. any particular reason they are 4 to a line?

Comment (virtuald) [reply wpilib/tests/test_drive.py]: Not really, but the formatting black applies is much worse IIRC.

File: wpilib/tests/test_pwm_motor_controllers.py [hunk 3]
```diff
@@ -5,40 +5,53 @@
...
+)
 def test_controller(wpilib, hal_data, clsname):
-    
+
     # create object/helper function
     obj = getattr(wpilib, clsname)(2)
-    
+
     # validate reporting is correct
     # TODO: remove the VictorSPX and DMC60 check once there is usage reporting
```
Comment (ariovistus) [inline wpilib/tests/test_pwm_motor_controllers.py]: out of scope for this pr, but these two have usage reporting now

File: wpilib/tests/test_shuffleboard.py [hunk 4]
```diff
@@ -23,7 +23,7 @@ def test_addPersistent(wpilib, getTable):
     tab = wpilib.shuffleboard.Shuffleboard.getTab("tacos")
     widget = tab.addPersistent("simplevalue", 1)
     # k, how to kick networktables?
-    #assert widget.getEntry().isPersistent()
+    # assert widget.getEntry().isPersistent()
```
Comment (ariovistus) [inline wpilib/tests/test_shuffleboard.py]: rats, did I forget to ask about this for the shuffleboard pr?

Comment (auscompgeek) [reply wpilib/tests/test_shuffleboard.py]: lol. that should be a thing, unless you forgot to make `getEntry` a thing.

File: hal-roborio/hal_impl/types.py [hunk 5]
```diff
@@ -1,3 +1,5 @@
+# fmt: off
```
Comment (auscompgeek) [inline hal-roborio/hal_impl/types.py]: Surely this doesn't look _that_ terrible after formatting.

Comment (virtuald) [reply hal-roborio/hal_impl/types.py]: It's not very good, as it's fairly abnormal code. And I'd rather it stay the way it is.

**Comment by virtuald** [comment]
All your pull request are belong to me.

**Comment by virtuald** [comment]
@ariovistus you'll need to rebase all your stuff. Sorry, but not sorry? In the long run, you'll thank me because @auscompgeek will find less things to nitpick about ;)

**Comment by auscompgeek** [comment]
yay I finally no longer have to care about style :stuck_out_tongue_winking_eye:

**Pull request merged**